include/spirv-tools/optimizer.hpp

```diff
@@ -956,6 +956,18 @@ Optimizer::PassToken CreateFixFuncCallArgumentsPass();
 // the unknown capability interacts with one of the trimmed capabilities.
 Optimizer::PassToken CreateTrimCapabilitiesPass();
 
+// Creates a struct-packing pass.
+// This pass re-assigns all offset layout decorators to tightly pack
+// the specified struct according to its packing rules.
+// This is especially useful when spirv-opt removes lots of unused fields
+// in HLSL's "$Globals" struct after which this pass can reduce its memory
+// footprint. Accepted packing rules are:
+// std140, std140EnhancedLayout, std430,
+// std430EnhancedLayout, hlslCbuffer, hlslCbufferPackOffset, scalar,
+// scalarEnhancedLayout.
```

Comment:
It is not clear what "structToPack" is supposed to be. Should it match the OpName for the struct? Also I personally prefer to keep the justification for the pass in the PR. The comment should describe what the function does. If so, this might be a better comment:

```suggestion
// Creates a struct-packing pass.
// This pass re-assigns all offset layout decorators to tightly pack
// the struct with OpName matching `structToPack` according to the given packing rule.
// Accepted packing rules are:
// std140, std140EnhancedLayout, std430,
// std430EnhancedLayout, hlslCbuffer, hlslCbufferPackOffset, scalar,
// scalarEnhancedLayout.
```